src/v/redpanda/admin_server.cc

```diff
@@ -2858,7 +2858,7 @@ void admin_server::register_cluster_routes() {
                       return result_t(backend.overview());
                   } else {
                       auto leader_id = backend.leader_id();
-                      if (leader_id) {
+                      if (leader_id && leader_id != config::node().node_id()) {
```

Comment:
I think moving this logic into `backend.leader_id()` is better - so that other possible clients don't have to repeat it.